sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,117 @@
+	clientSecret *string // Gets the client secret that was generated for the App Registration used to authenticate the client.
+}
+
+// NewInteractiveBrowserCredential constructs a new InteractiveBrowserCredential with the details needed to authenticate against Azure Active Directory through an interactive browser window.
+// tenantID: The Azure Active Directory tenant (directory) ID of the service principal.
+// clientID: The client (application) ID of the service principal.
+// clientSecret: Gets the client secret that was generated for the App Registration used to authenticate the client.
+// options: allow to configure the management of the requests sent to Azure Active Directory.
+func NewInteractiveBrowserCredential(tenantID string, clientID string, options *InteractiveBrowserCredentialOptions) (*InteractiveBrowserCredential, error) {
+	c, err := newAADIdentityClient(options.Options)
+	if err != nil {
+		return nil, err
+	}
+	var clientSecret *string
+	if options.ClientSecret != nil {
+		clientSecret = options.ClientSecret
+	}
+	return &InteractiveBrowserCredential{tenantID: tenantID, clientID: clientID, clientSecret: clientSecret, client: c}, nil
+}
+
+// GetToken obtains a token from Azure Active Directory using an interactive browser to authenticate.
+// ctx: Context used to control the request lifetime.
+// opts: TokenRequestOptions contains the list of scopes for which the token will have access.
+// Returns an AccessToken which can be used to authenticate service client calls.
+func (c *InteractiveBrowserCredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	tk, err := c.client.authenticateInteractiveBrowser(ctx, c.tenantID, c.clientID, c.clientSecret, opts.Scopes)
+	if err != nil {
+		addGetTokenFailureLogs("Interactive Browser Credential", err, true)
+		return nil, err
+	}
+	logGetTokenSuccess(c, opts)
+	return tk, nil
+}
+
+// AuthenticationPolicy implements the azcore.Credential interface on InteractiveBrowserCredential and calls the Bearer Token policy
+// to get the bearer token.
+func (c *InteractiveBrowserCredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	return newBearerTokenPolicy(c, options)
+}
+
+var _ azcore.TokenCredential = (*InteractiveBrowserCredential)(nil)
+
+var authCodeReceiver = func(tenantID string, clientID string, scopes []string) (*interactiveConfig, error) {
+	return interactiveBrowserLogin(tenantID, clientID, scopes)
+}
+
+// interactiveBrowserLogin opens an interactive browser with the specified tenant and client IDs provided then returns the authorization code
+// received or an error.
+func interactiveBrowserLogin(tenantID string, clientID string, scopes []string) (*interactiveConfig, error) {
+	const authURLFormat = "https://login.microsoftonline.com/%s/oauth2/v2.0/authorize?response_type=code&response_mode=query&client_id=%s&redirect_uri=%s&state=%s&scope=%s&prompt=select_account"
```

Comment:
The authority host should be configurable, this only will work for public cloud.

Reply:
Also done